REGRESSION (276531@main): State dropdown is invisible, unable to send feedback via Bunnings.com.au

nt1m opened the pull request:
#### 72ac253c4d48099635aa14a321e0c7e1ca267cb2
<pre>
REGRESSION (276531@main): State dropdown is invisible, unable to send feedback via Bunnings.com.au
<a href="https://bugs.webkit.org/show_bug.cgi?id=277303">https://bugs.webkit.org/show_bug.cgi?id=277303</a>
<a href="https://rdar.apple.com/130520487">rdar://130520487</a>

Reviewed by Cameron McCormack.

We should not be cancelling animations in the case where we&apos;re starting in a hidden subtree, then animating with a keyframe that changes
the display value away from none. Make sure we notify about the newly created keyframe instead.

* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/display-none-dont-cancel.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/display-none-dont-cancel.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/display-none-to-display-block-dont-cancel.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/display-none-to-display-block-dont-cancel.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/display-none-to-display-block-expected.xht: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/display-none-to-display-block.html: Added.
* Source/WebCore/style/StyleTreeResolver.cpp:
(WebCore::Style::TreeResolver::createAnimatedElementUpdate):

Canonical link: <a href="https://commits.webkit.org/281544@main">https://commits.webkit.org/281544@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2229a7b7b5b1c49e3c84391381cb61749ae000a1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60270 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39622 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/12828 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64190 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/10802 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/62400 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47294 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11035 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/48803 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7515 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62301 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/36931 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52208 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29645 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33630 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/9437 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/9719 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/55529 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9725 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/65922 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4204 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/9576 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/56159 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4222 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52185 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56320 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13367 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3494 "Passed tests") | | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/35432 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/36513 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/37603 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/36257 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

heycam commented on LayoutTests/imported/w3c/web-platform-tests/css/css-animations/display-none-dont-cancel.tentative.html:
```diff
@@ -72,7 +76,8 @@
 <script>
-promise_test(async () => {
+promise_test(async (t) => {
+  t.add_cleanup(() => target3.classList.remove('animate3'));
```
Nit: you added a blank line after this in the previous subtests. Let's be consistent about it in this and the later subtests.

heycam replied:
Also: do we still need to make these changes in this test, if the new test you're adding is in a separate file?

nt1m replied:
It's nice to cleanup after tests either way, though not needed.

heycam commented on LayoutTests/imported/w3c/web-platform-tests/css/css-animations/display-none-to-display-block-dont-cancel.tentative.html:
```diff
@@ -0,0 +1,41 @@
+</style>
+<script>
+promise_test(async (t) => {
+  t.add_cleanup(() => target.classList.remove('animate'));
```
Do we need this?

nt1m replied:
Same, it's name to reset the state of the test so they can be re-ran if needed when opened manually

heycam commented on LayoutTests/imported/w3c/web-platform-tests/css/css-animations/display-none-to-display-block-dont-cancel.tentative-expected.txt:
```diff
@@ -0,0 +1,3 @@
+
+PASS display:none animating to display:block should be block for the whole animation.
```
Does this expectation file need updating? It doesn't have the output from the "Only one animation should start." assertion.

nt1m replied:
That's how the subtest is named.

heycam reviewed (approved):
r=me with comments addressed.

nt1m committed:
REGRESSION (276531@main): State dropdown is invisible, unable to send feedback via Bunnings.com.au
https://bugs.webkit.org/show_bug.cgi?id=277303
rdar://130520487

Reviewed by Cameron McCormack.

We should not be cancelling animations in the case where we're starting in a hidden subtree, then animating with a keyframe that changes
the display value away from none. Make sure we notify about the newly created keyframe instead.

* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/display-none-dont-cancel.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/display-none-dont-cancel.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/display-none-to-display-block-dont-cancel.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/display-none-to-display-block-dont-cancel.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/display-none-to-display-block-expected.xht: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/display-none-to-display-block.html: Added.
* Source/WebCore/style/StyleTreeResolver.cpp:
(WebCore::Style::TreeResolver::createAnimatedElementUpdate):

Canonical link: https://commits.webkit.org/281544@main
